SQL/4주차/SB_취소되지_않은_진료_예약_조회하기.sql

```diff
@@ -0,0 +1,11 @@
+-- 코드를 입력하세요
+-- 2022년 4월 13일 취소되지 않은 흉부외과(CS) 진료 예약 내역을 조회
+SELECT a.APNT_NO, p.PT_NAME, p.PT_NO, a.MCDP_CD, d.DR_NAME, a.APNT_YMD
+FROM APPOINTMENT a
+JOIN DOCTOR d ON a.MDDR_ID = d.DR_ID
+JOIN PATIENT p ON a.PT_NO = p.PT_NO
+WHERE 
+    DATE_FORMAT(a.APNT_YMD, '%Y-%m-%d') = '2022-04-13'
```

Comment:
만약에 `APNT_YMD` 가 인덱스가 걸려있다고 한다면 `DATE_FORMAT()` 함수보다 `LIKE` 의 와일드카드를 끝에 뒤는게 성능상 더 좋아요!
SQL표준으로 `WHERE`절의 왼쪽에 함수나 연산자가 위치할 때 인덱스를 사용이 불가해서 `DATE_FORMAT(a.APNT_YMD)`가 왼쪽에 위치 하기 때문에 인덱스 조회가 불가능하다고 해요

사실 이 문제는 인덱스 유무 여부는 모르지만, 저였으면 `LIKE` 를 사용했을 것 같은데 그럼 성능상 어떤 차이가 있을까 찾아보다가 코멘트 남깁니다 ㅎㅎ
```suggestion
WHERE 
    a.APNT_YMD LIKE '2022-04-13%'
```

- 참고
[LIKE 와일드카드 별 인덱싱](https://use-the-index-luke.com/sql/where-clause/searching-for-ranges/like-performance-tuning)
[Mysql 성능 개선](https://dbguide.tistory.com/entry/MySQL-%EC%84%B1%EB%8A%A5-%EC%A3%BD%EC%9D%B4%EB%8A%94-%EC%9E%98%EB%AA%BB%EB%90%9C-%EC%BF%BC%EB%A6%AC-%EC%8A%B5%EA%B4%80)

Reply:
아하! WHERE절의 왼쪽에 함수나 연산자가 오면 인덱스를 사용할 수 없고, 그렇기에 성능저하가 일어날 수 있기에 인덱스를 사용할 수 있는 LIKE연산자를 이용하는게 더 효율적이군요!!
SQL모르는거 투성이인데 예진님 덕분에 하나씩 배워갑니다!! 다음부터 문제 풀때 고려하면서 풀어보겠습니다!👍👍